### DISCUSSION TITLE
Inconsistent Z offset changes

### DISCUSSION THREAD

**Discussion by dwery** [Jyers/Marlin#986] [Q&A] [answered]
Following on the issue https://github.com/Jyers/Marlin/issues/979 , here are my tests.

Mesh is created with `M851 Z-0.92` , `G29 P1` and `G29 P3`. My G28 restores the leveling state. Visual inspection of the mesh looks good.

Start G-Code:

```
M851 Z-0.92
G29 L0
M420 S1
G28
```
When printing starts, I use the tune menu to bring the offset to -0.78 using the direct value input, not the microstep up/down,  and I get a good first layer. The same happens if I use  `M290 +0.14`.

If I change that G-Code to `M851 Z-0.78` I'd expect the same results. However the nozzle is clearly more into the bed and the first layer has the typical scratches of that situation. This is 100% reproducible.

**Comment by Jyers** [comment, marked as answer]
It is somewhat expected for there to some inconsistency in large Z offset changes during printing. This is mostly due to the fact that these adjustments are done by babystepping, which I'm sure you know is a very low torque operation. This means while it should be fairly accurate in the scope of +- 0.05 mm, I could definitely see 0.14 not making it the full distance. This should be easily ironed out after either adjusting it directly in the wizard and using the re-home z to check, or continuing to adjust it each print as the amount you need to change it should get consistently smaller.

**Reply by dwery** [reply]
I’m verging toward a theory where Z offset changes in M851 are only accounted for if the probe is used for z homing (which I’m not doing yet). I’ll have to check if UBL keeps track of the offset that has been used when creating the mesh or not.

**Reply by Jyers** [reply]
Interesting, maybe some relation to #557 then

**Reply by dwery** [reply]
Sounds very similar, yes.
